src/governance-classes.cpp

```diff
@@ -0,0 +1,492 @@
+            continue;
+        }
+        
+        // DO WE HAVE A NEW WINNER?
+
+        int nTempYesCount = pObj->GetAbsoluteYesCount(VOTE_SIGNAL_FUNDING);
+        DBG( cout << "GetBestSuperblock nTempYesCount = " << nTempYesCount << endl; );
+        if(nTempYesCount > nYesCount)  {
+            nYesCount = nTempYesCount;
+            pBlock = superblock;
+            DBG( cout << "GetBestSuperblock Valid superblock found, pBlock set" << endl; );
+        }       
+    }
+
+    return nYesCount > 0;
+}
+
+/**
+*   Create Superblock Payments
+*
+*   - Create the correct payment structure for a given superblock
+*/
+
+void CSuperblockManager::CreateSuperblock(CMutableTransaction& txNew, CAmount nFees, int nBlockHeight)
+{
+    DBG( cout << "CSuperblockManager::CreateSuperblock Start" << endl; );
+
+    LOCK(governance.cs);
+    AssertLockHeld(cs_main);
+
+    if(!chainActive.Tip()) {
+        DBG( cout << "CSuperblockManager::CreateSuperblock No active tip, returning" << endl; );
+        return;
+    }
+
+    // GET THE BEST SUPERBLOCK FOR THIS BLOCK HEIGHT
+
+    CSuperblock_sptr pBlock;
+    if(!CSuperblockManager::GetBestSuperblock(pBlock, nBlockHeight))  {
+        LogPrint("superblock", "CSuperblockManager::CreateSuperblock: Can't find superblock for height %d\n", nBlockHeight);
+        DBG( cout << "CSuperblockManager::CreateSuperblock Failed to get superblock for height, returning" << endl; );
+        return;
+    }
+
+    // CONFIGURE SUPERBLOCK OUTPUTS 
+
+    DBG( cout << "CSuperblockManager::CreateSuperblock Number payments: " << pBlock->CountPayments() << endl; );
+
+    txNew.vout.resize(pBlock->CountPayments());
+    for(int i = 0; i <= pBlock->CountPayments(); i++)  {
+        CGovernancePayment payment;
+        DBG( cout << "CSuperblockManager::CreateSuperblock i = " << i << endl; );
+        if(pBlock->GetPayment(i, payment))  {
+            DBG( cout << "CSuperblockManager::CreateSuperblock Payment found " << endl; );
+            // SET COINBASE OUTPUT TO SUPERBLOCK SETTING
+            
+            txNew.vout[i].scriptPubKey = payment.script;
+            txNew.vout[i].nValue = payment.nAmount;
+            
+            // PRINT NICE LOG OUTPUT FOR SUPERBLOCK PAYMENT
+            
+            CTxDestination address1;
+            ExtractDestination(payment.script, address1);
+            CBitcoinAddress address2(address1);
+            
+            // TODO: PRINT NICE N.N DASH OUTPUT
+            
+            DBG( cout << "CSuperblockManager::CreateSuperblock Before LogPrintf call " << endl; );
+            LogPrintf("NEW Superblock : output %d (addr %s, amount %d)\n", i, address2.ToString(), payment.nAmount);
+            DBG( cout << "CSuperblockManager::CreateSuperblock After LogPrintf call " << endl; );
+            pBlock->SetExecuted();
+        }
+        else  {
+            DBG( cout << "CSuperblockManager::CreateSuperblock Payment not found " << endl; );
```

Comment:
No, no. I probably don't quite understand the whole scheme of triggers etc but it's kind of weird that we have valid superblock, we know number of payments in it (`pBlock->CountPayments()`) and yet we were not able to get `i`th payment from that list.